clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/ScaleCloudFoundryServerGroupAtomicOperation.java

```diff
@@ -56,17 +57,20 @@ public Void operate(List priorOutputs) {
 
     ProcessStats.State state = operationPoller.waitForOperation(
       () -> client.getApplications().getProcessState(description.getServerGroupId()),
-      inProgressState -> (inProgressState == ProcessStats.State.RUNNING || inProgressState == ProcessStats.State.CRASHED),
+      inProgressState -> (
+        inProgressState == ProcessStats.State.RUNNING
+          || inProgressState == ProcessStats.State.CRASHED
+          || inProgressState == ProcessStats.State.DOWN),
       null, getTask(), description.getServerGroupName(), PHASE);
 
-    if (state != ProcessStats.State.RUNNING) {
+    if (state == ProcessStats.State.RUNNING || (state == ProcessStats.State.DOWN &&
+      Optional.ofNullable(description.getCapacity()).map(cap -> cap.getDesired() == 0).orElse(false))) {
```

Comment:
What is the case where capacity can be null?

Reply:
We took a closer look and Capacity should not be null because it will either be provided from `Orca` or specified from the user in `Deck`